src/Uno.XamlMerge.Task/BatchMergeXaml.cs

```diff
@@ -119,12 +151,151 @@ class BatchMerger
 
                 var projectBasePath = Path.GetDirectoryName(Path.GetFullPath(projectFullPath));
 
+                var dictionary = new Dictionary<string, string>();
+                var documents = new List<XmlDocument>();
+
+                // This dictionary handles elements, e.g, <android:MyAndroid />, where the key is the XElement
+                // and the value is the prefix (e.g, "android")
+                var elementsToUpdate = new Dictionary<XmlElement, string>();
+
+                // This dictionary handles attributes that are namespace declarations, e.g xmlns:android="..."
+                // where the key is the XAttribute and the value is the namespace name (e.g, "android")
+                var attributesToUpdate = new Dictionary<XmlAttribute, string>();
+
+                // This dictionary handles attributes that are property prefixes, e.g, <MyElement android:MyProp="Value" />
+                var propertyAttributesToUpdate = new Dictionary<XmlAttribute, string>();
+
+                
                 foreach (string page in pages)
                 {
+                    try
+                    {
+                        var document = new XmlDocument();
+                        document.Load(page);
+
+                        foreach (XmlNode node in document.SelectNodes("descendant::node()"))
+                        {
+                            if (node is XmlElement element)
+                            {
+                                var prefix = element.GetPrefixOfNamespace(element.NamespaceURI);
+                                if (prefix is "xamarin" or "not_win" or "android" or "ios" or "wasm" or "macos" or "skia")
+                                {
+                                    elementsToUpdate.Add(element, prefix);
+                                }
+
+                                foreach (XmlAttribute att in element.Attributes)
+                                {
+                                    if (att.Name.StartsWith("xmlns:"))
+                                    {
+                                        string name = att.LocalName;
+                                        if (name is "xamarin" or "not_win" or "android" or "ios" or "wasm" or "macos" or "skia")
```

Comment:
Could casing matter here? I am not sure if `xmlns:iOS`= is also valid or not

Reply:
I verified that this is case-sensitive. So it should remain.

Reply:
I'm not sure we should be supporting those namespaces in a hard-coded way. Any of those namespaces will be in the `ignorable` attribute and may be able to use that list instead?

Developers may add custom ones, and we will certainly add more in the future.

Reply:
@jeromelaban I went with a slightly different approach, instead of using `ignorable` attribute, I relied on `IncludeXamlNamespaces` and `ExcludeXamlNamespaces` MSBuild items. But let me know if my approach isn't correct.

Reply:
That works as well. Thanks!